src/app/app.component.ts

```diff
@@ -80,9 +84,52 @@ export class AppComponent implements OnInit {
       });
   }
 
+  toggle(value: string) {
+    if (value === 'toggle' && !this.window.isXS) {
+      this.sideMenu = !this.sideMenu;
+      this.unPin = this.sideMenu ? 'Hide Menu' : 'Show Menu';
+      this.resizeMenu('toggle');
+      return;
+    }
+    if (!this.sideMenu) {
+      this.isPinned = value === 'enter';
+    }
+  }
+
+  resizeMenu(type: string) {
+    const elem = document.getElementsByClassName('mat-drawer-content') as HTMLCollectionOf<HTMLElement>;
+    let left;
+    if (type === 'toggle') {
+      left = elem[0].offsetLeft === 100 ? 250 : 100;
+    } else if (type === 'resize') {
+        if (this.window.isXS) {
+          this.showNav = false;
```

Comment:
This will always be true, and left will always be 0

Reply:
yeah, I think that condition is not required 👍